### ISSUE TITLE
Guide button does not work

### ISSUE THREAD

**Issue by hifihedgehog** [x360ce/x360ce#1223]
See above. In version 4.17.15.0, the guide button does not register no matter what device and button I map to it. The record step detects input and seems to finish recording, but the button itself does not work. All other buttons and axes work.

**Comment by hifihedgehog** [comment]
Relates to issue #1203

**Comment by hifihedgehog** [comment]
If the Guide button is actually working, it should register as button B11 if I am not mistaken on gamepad-tester.com.

**Comment by hifihedgehog** [comment]
XOutput uses the same ViGEm interface as x360ce and it simulates a true Guide button, not a virtual key. See code here:

https://github.com/csutorasa/XOutput/blob/f784c48091f3657f8ecbcd9f87901756ff4967c7/XOutput.Emulation/ViGEm/VigemXboxDevice.cs

**Comment by hifihedgehog** [comment]
Please add the following line of code to **x360ce/x360ce.App.Beta/Common/DInput/DInputHelper.Step5.VirtualDevices.cs** directly after **line 146** 

> report.SetButtonState(Xbox360Buttons.Guide, n.Buttons.HasFlag(GamepadButtonFlags.Guide));

Add it after here:
https://github.com/x360ce/x360ce/blob/f295e72ea242ca180b5a09389d63eb3fd503d995/x360ce.App.Beta/Common/DInput/DInputHelper.Step5.VirtualDevices.cs#L146 


It should enable true Guide button functionality since ViGEm fully supports the Guide button and XOutput uses it (see here: https://github.com/csutorasa/XOutput/blob/f784c48091f3657f8ecbcd9f87901756ff4967c7/XOutput.Emulation/ViGEm/VigemXboxDevice.cs#L61). It is cool that you can use the keyboard mapping instead of a true Guide button, which is good in some casse, but many games still expect a true Guide button. 

Please fix this. Thanks!

**Comment by hifihedgehog** [comment]
@EJocys Did you get an opportunity to see this issue and my proposed fix?

**Comment by EJocys** [comment]
@hifihedgehog Yes, I've read your report. Thank you. I will return to map issue fixing after I finish upgrade of interface from obsolete Windows Forms to more modern WPF (XAML), which will resolve issues of app when running on high-res monitors. During last 3 months converted 36 forms. 13 to go. So, it will take about another month to finish. Then maybe week for tests and stabilization of the app with rewritten interface. Then I can focus again on Bluetooth, Guide and other mapping issues.

**Comment by sentix001** [comment]
was this ever sorted?

**Comment by hifihedgehog** [comment]
> was this ever sorted?

No. This project seems to be in limbo. I hope they get another dev on board.

**Comment by EvoJocys** [comment]
Fixed. From the changelog:

4.17.22 — Fixed: Guide button sends correct virtual key code; action configurable.
4.17.46 — Fixed: Guide button was not passed to the virtual controller.

Download: https://github.com/x360ce/x360ce/releases

Closing as part of a tracker clean-up after the 4.21 release. If it still happens on 4.21, reopen this issue with the version and what you see.
